lib/pure/asyncfutures.nim

```diff
@@ -30,7 +34,14 @@ type
 when not defined(release):
   var currentID = 0
 
-proc callSoon*(cbproc: proc ()) {.gcsafe.}
+var callSoonProc* {.threadvar.}: (proc(cbproc: proc ()) {.gcsafe.})
```

Comment:
Ok.

Interfaces won't help here - asyncfutures only needs this single callSoon proc and I think it shouldn't depend on whole AsyncLoop interface. (anyway, I think there are better options than using interface for AsyncLoop, but that's another topic)